iframe.html

```diff
@@ -13,62 +13,66 @@
-    } else {
-      console.log("iframe.html - document.referrer was not set");
-    }
+    function queryStringToJSON(qs) {
+      var pairs = qs.split('&');
+      var result = {};
+      pairs.forEach(function(pair) {
+        var pair = pair.split('=');
+        var key = pair[0];
+        var value = decodeURIComponent(pair[1] || '');
+        if (value === "true") {
+          value = true;
+        } else if (value == "false") {
+          value = false;
+        } else if (!isNaN(parseFloat(value))) {
+          value = parseFloat(value);
+        } else if (value.match(/^\[.+\]$/)) {
+          value = JSON.parse(value);
+        } else if (value.match(/^\{.+\}$/)) {
+          value = JSON.parse(value);
+        } 
 
-    // Parse standard Ooyala player runtime options
-    var queryTokens = queryParams.split(/[\?\&]/);
-    for (var i=0; i<queryTokens.length; i++) {
-      try {
-        var match = queryTokens[i].match(/^options\[([^\]]+)\]=(.+)/);
-        if (match) {
-          var currentLevel = playerParam;
-          var keySplits = match[1].split(".");
-          while (keySplits.length > 1) {
-            var key = keySplits.shift();
-            if (currentLevel[key] === undefined)
-              currentLevel[key] = {};
-            currentLevel = currentLevel[key];
+        if (result[key]) {
+          if (Array.isArray(result[key])) {
+            result[key].push(value);
+          } else {
+            result[key] = [result[key], value];
           }
-          if (typeof currentLevel !== 'object')
-            throw("Exception: Invalid query parameter");
+        } else {
+          result[key] = value;
+        }
+      });
 
-          var paramValue = decodeURIComponent(match[2]);
-          if (paramValue.toLowerCase() === "true")
-            currentLevel[keySplits[0]] = true;
-          else if (paramValue.toLowerCase() === "false")
-            currentLevel[keySplits[0]] = false;
-          else if (paramValue[0] === "[" &&
-            paramValue[paramValue.length -1] === "]")
-            currentLevel[keySplits[0]] = JSON.parse(decodeURIComponent(match[2]));
-          else
-            currentLevel[keySplits[0]] = decodeURIComponent(match[2]);
+      return JSON.parse(JSON.stringify(result));
+    };
+   
+    var playerParams = queryStringToJSON(window.location.search.substring(1));
+    var skin = {
+      config: playerParams["skin.config"] || "",
+      inline: {
+        shareScreen: {
+          embed: {
+            source: "<iframe width='640' height='480' frameborder='0' allowfullscreen src='//player.ooyala.com/static/v4/stable/latest/skin-plugin/iframe.html?ec=<ASSET_ID>&pbid=<PLAYER_ID>&pcode=<PUBLISHER_ID>'></iframe>"
+          }
         }
       }
-      catch (e) {
-        throw("Exception: Invalid query parameter");
+    };
+    playerParams["playerBrandingId"] = playerParams.pbid;
+    playerParams["skin"] = skin;
+    delete playerParams.pbid;
```

Comment:
We may leave it, but this will be unused parameter, it's cleaner this way I believe.